lfsapi/creds.go

```diff
@@ -90,76 +90,72 @@ type AskPassCredentialHelper struct {
 // The ASKPASS program is only queried if a credential was not already
 // provided, i.e. through the git URL
 func (a *AskPassCredentialHelper) Fill(what Creds) (Creds, error) {
-	var user bytes.Buffer
-	var pass bytes.Buffer
-	var err bytes.Buffer
-
-	var username string
-	var password string
-
 	u := &url.URL{
 		Scheme: what["protocol"],
 		Host:   what["host"],
 		Path:   what["path"],
 	}
 
+	creds := make(Creds)
+
 	if givenUser, ok := what["username"]; ok {
-		username = givenUser
+		creds["username"] = givenUser
 	} else {
-		// 'ucmd' will run the GIT_ASKPASS (or core.askpass) command prompting
-		// for a username.
-		ucmd := exec.Command(a.Program, a.args(fmt.Sprintf("Username for %q", u))...)
-		ucmd.Stderr = &err
-		ucmd.Stdout = &user
-
-		tracerx.Printf("creds: filling with GIT_ASKPASS: %s", strings.Join(ucmd.Args, " "))
-		if err := ucmd.Run(); err != nil {
+		username, err := a.getFromProgram("Username", u)
+		if err != nil {
 			return nil, err
 		}
 
-		if err.Len() > 0 {
-			return nil, errors.New(err.String())
-		}
-		username = user.String()
+		creds["username"] = username
 	}
 
-	username = strings.TrimSpace(username)
-	if len(username) > 0 {
+	if len(creds["username"]) > 0 {
 		// If a non-empty username was given, add it to the URL via func
 		// 'net/url.User()'.
-		u.User = url.User(username)
+		u.User = url.User(creds["username"])
 	}
 
 	if givenPass, ok := what["password"]; ok {
-		password = givenPass
+		creds["password"] = givenPass
 	} else {
-		// Create 'pcmd' to run the GIT_ASKPASS (or core.askpass)
-		// command prompting for a password.
-		pcmd := exec.Command(a.Program, a.args(fmt.Sprintf("Password for %q", u))...)
-		pcmd.Stderr = &err
-		pcmd.Stdout = &pass
-
-		tracerx.Printf("creds: filling with GIT_ASKPASS: %s", strings.Join(pcmd.Args, " "))
-		if err := pcmd.Run(); err != nil {
+		password, err := a.getFromProgram("Password", u)
+		if err != nil {
 			return nil, err
 		}
 
-		if err.Len() > 0 {
-			return nil, errors.New(err.String())
-		}
-
-		password = pass.String()
+		creds["password"] = password
 	}
 
-	// Finally, now that we have the username and password information,
-	// store it in the creds instance that we will return to the caller.
-	creds := make(Creds)
-	creds["username"] = username
-	creds["password"] = strings.TrimSpace(password)
+	creds["username"] = strings.TrimSpace(creds["username"])
```

Comment:
I think these final `strings.TrimSpace` calls are unnecessary, since the body of `getFromProgram` calls strings.TrimFromSpace` (and its an invariant that `what` will have space-less values, too).